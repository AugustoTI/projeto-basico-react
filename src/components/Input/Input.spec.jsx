import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { Input } from './index';

describe('<Input />', () => {
    it('should have a value of valueInput', () => {
        const fn = jest.fn();
        render(<Input valueInput={'testando'} action={fn} />);

        const input = screen.getByPlaceholderText(/Type your Search/i);
        expect(input.value).toBe('testando');
    });

    it('should call handleChange function on each key press', () => {
        const fn = jest.fn();
        render(<Input action={fn} />);

        const input = screen.getByPlaceholderText(/Type your Search/i);
        const value = 'o valor digitado';

        userEvent.type(input, value);

        expect(input.value).toBe(value);
        expect(fn).toHaveBeenCalledTimes(value.length);
    });

    it('should macth snapshot', () => {
        const fn = jest.fn();
        const { container } = render(<Input action={fn} />);
        expect(container.firstChild).toMatchSnapshot();
    });
});
