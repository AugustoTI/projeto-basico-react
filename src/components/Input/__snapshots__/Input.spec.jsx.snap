// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should macth snapshot 1`] = `
<input
  class="search-input"
  placeholder="Type your Search"
  type="search"
  value=""
/>
`;
